analyzers/src/SonarAnalyzer.CSharp/Rules/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -23,131 +23,119 @@ namespace SonarAnalyzer.Rules.CSharp
-                if (returnNullStatements.Count > 0 &&
-                    IsReturningCollection(context))
+
+                if (nullOrDefaultLiterals.Count > 0 && IsReturningCollection(context))
                 {
-                    context.ReportIssue(rule.CreateDiagnostic(context.Compilation, returnNullStatements[0], additionalLocations: returnNullStatements.Skip(1)));
+                    context.ReportIssue(Rule.CreateDiagnostic(context.Compilation, nullOrDefaultLiterals[0], additionalLocations: nullOrDefaultLiterals.Skip(1)));
                 }
             }
         }
 
         private static bool IsReturningCollection(SyntaxNodeAnalysisContext context)
         {
             var symbol = context.SemanticModel.GetDeclaredSymbol(context.Node);
-
             var methodSymbol = (symbol as IPropertySymbol)?.GetMethod ?? symbol as IMethodSymbol;
 
-            return methodSymbol != null &&
-                !methodSymbol.ReturnType.Is(KnownType.System_String) &&
-                !methodSymbol.ReturnType.DerivesFrom(KnownType.System_Xml_XmlNode) &&
-                methodSymbol.ReturnType.DerivesOrImplementsAny(CollectionTypes);
+            return methodSymbol != null
+                && !methodSymbol.ReturnType.Is(KnownType.System_String)
+                && !methodSymbol.ReturnType.DerivesFrom(KnownType.System_Xml_XmlNode)
+                && methodSymbol.ReturnType.DerivesOrImplementsAny(CollectionTypes);
         }
 
-        private static ArrowExpressionClauseSyntax GetExpressionBody(SyntaxNode node)
-        {
-            switch (node.Kind())
+        private static ArrowExpressionClauseSyntax GetExpressionBody(SyntaxNode node) =>
+            node switch
             {
-                case SyntaxKind.MethodDeclaration:
-                    return ((MethodDeclarationSyntax)node).ExpressionBody;
-
-                case SyntaxKind.PropertyDeclaration:
-                    var property = (PropertyDeclarationSyntax)node;
-
-                    if (property.ExpressionBody != null)
-                    {
-                        return property.ExpressionBody;
-                    }
-
-                    return property.AccessorList?.Accessors
-                        .FirstOrDefault(accessor => accessor.IsKind(SyntaxKind.GetAccessorDeclaration))
-                        ?.ExpressionBody();
+                MethodDeclarationSyntax method => method.ExpressionBody,
+                PropertyDeclarationSyntax property => property.ExpressionBody ?? GetGetAccessor(property)?.ExpressionBody(),
+                var _ when LocalFunctionStatementSyntaxWrapper.IsInstance(node) => ((LocalFunctionStatementSyntaxWrapper)node).ExpressionBody,
+                OperatorDeclarationSyntax @operator => @operator.ExpressionBody,
+                _ => null,
+            };
+
+        private static BlockSyntax GetBody(SyntaxNode node) =>
+            node switch
+            {
+                MethodDeclarationSyntax method => method.Body,
```

Comment:
You might try using `BaseMethodDeclarationSyntax` which will cover both `MethodDeclarationSyntax` `OperatorDeclarationSyntax`  cases and will simplify the changes to support `ConversionOperatorDeclarationSyntax`.